wordpress-mobile/WordPress-Android
# [Notifications Refresh] 🤖 Milestone Details: Badge and Title

## Description
This PR:
* Refactors the code separating the milestone implementation
* Removes deprecated code
* Partially implements https://github.com/Automattic/wordpress-mobile/issues/23:
  3️⃣ Set the achievement icon
  4️⃣ Set the achievement text
The rest of the items will be covered in follow up PRs.

-----

## To Test:
1. Open the notifications tab
2. Open a milestone notification
3. *Verify* that the badge and text is displayed as expected
4. Browse other notification types
5. *Verify* that the functionality is unchanged

-----

## Regression Notes

1. Potential unintended areas of impact

    - Milestone notifications

6. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual testing

7. What automated tests I added (or what prevented me from doing so)

    - Will iterate on a follow up PR

-----

## PR Submission Checklist:


-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):
